### ISSUE TITLE
update to nc-aio 10 from nc-aio 8.x failed (nextcloud-aio-nextcloud won't start)

### ISSUE THREAD

**Issue by combrs** [nextcloud/all-in-one#5683]
I use docker-installed nextcloud-aio and follow standard upgrade path to 10.0 via https://myhost:11080/api ....token=5ad2e0f...
mastercontainer was upgraded successfully
containers start to upgrade, redis & db upgraded successfully and started
others are depend on nextcloud-aio-nextcloud, but it was not started (waiting to start in loop, but cannot start because " update.failed file in the datadir " was created

I was wait for half of hour, bit see in log only "failed message" loop. Then I delete "update.failed" file and restart container  nextcloud-aio-nextcloud, and it _continue_ to update (and finish it successfully). Logs below

### Steps to reproduce
1.  Try to upgrade?

### Expected behavior 

Smooth upgrade as earlier versions

### Actual behavior 

Upgrade hangs and containers was not start

#### Output of `sudo docker info`
Server Version: 20.10.23
 OSType: linux
 Architecture: x86_64
 CPUs: 2
 Security Options:
  apparmor
 Kernel Version: 4.4.302+
 Operating System: Synology NAS
 Storage Driver: btrfs
  Build Version: Btrfs v4.0
  Library Version: 101
 Logging Driver: db
 Cgroup Driver: cgroupfs
 Cgroup Version: 1

#### Other valuable info 
upgrade failed log (see "Parse error: syntax error, unexpected token "/"" )

2024-11-30T10:06:49.430492541Z Connection to nextcloud-aio-database (172.18.0.3) 5432 port [tcp/postgresql] succeeded!
2024-11-30T10:06:52.240996762Z               now
2024-11-30T10:06:52.241438900Z -------------------------------
2024-11-30T10:06:52.241479283Z  2024-11-30 14:06:52.225645+04
2024-11-30T10:06:52.241518704Z (1 row)
2024-11-30T10:06:52.241551684Z
2024-11-30T10:06:52.339465137Z + '[' -f /dev-dri-group-was-added ']'
2024-11-30T10:06:52.339759934Z ++ find /dev -maxdepth 1 -mindepth 1 -name dri
2024-11-30T10:06:52.350560599Z + '[' -n '' ']'
2024-11-30T10:06:52.350628848Z + set +x
2024-11-30T10:06:52.404613611Z Enabling Imagick...
2024-11-30T10:07:00.501831372Z WARNING: opening from cache https://dl-cdn.alpinelinux.org/alpine/v3.20/main: No such file or directory
2024-11-30T10:07:00.502241053Z WARNING: opening from cache https://dl-cdn.alpinelinux.org/alpine/v3.20/community: No such file or directory
2024-11-30T10:07:00.628945907Z Connection to nextcloud-aio-redis (172.18.0.4) 6379 port [tcp/redis] succeeded!
2024-11-30T10:07:02.064018222Z + NEXT_MAJOR=29
2024-11-30T10:07:02.064309781Z + curl -fsSL -o nextcloud.tar.bz2 https://download.nextcloud.com/server/releases/latest-29.tar.bz2
2024-11-30T10:07:58.840349394Z + curl -fsSL -o nextcloud.tar.bz2.asc https://download.nextcloud.com/server/releases/latest-29.tar.bz2.asc
2024-11-30T10:07:59.804160355Z ++ mktemp -d
2024-11-30T10:08:00.203165398Z + GNUPGHOME=/tmp/tmp.c6uEYhj0E0
2024-11-30T10:08:00.203289590Z + export GNUPGHOME
2024-11-30T10:08:00.203320684Z + gpg --batch --keyserver keyserver.ubuntu.com --recv-keys 28806A878AE423A28372792ED75899B9A724937A
2024-11-30T10:08:00.960621648Z gpg: keybox '/tmp/tmp.c6uEYhj0E0/pubring.kbx' created
2024-11-30T10:08:02.270380410Z gpg: /tmp/tmp.c6uEYhj0E0/trustdb.gpg: trustdb created
2024-11-30T10:08:02.270854691Z gpg: key D75899B9A724937A: public key "Nextcloud Security <security@nextcloud.com>" imported
2024-11-30T10:08:02.693998851Z gpg: Total number processed: 1
2024-11-30T10:08:02.694156097Z gpg:               imported: 1
2024-11-30T10:08:02.694322899Z + gpg --batch --verify nextcloud.tar.bz2.asc nextcloud.tar.bz2
2024-11-30T10:08:04.524019459Z gpg: Signature made Thu Oct 10 13:18:34 2024 +04
2024-11-30T10:08:04.524453015Z gpg:                using RSA key 28806A878AE423A28372792ED75899B9A724937A
2024-11-30T10:08:04.525354312Z gpg: Good signature from "Nextcloud Security <security@nextcloud.com>" [unknown]
2024-11-30T10:08:04.525407498Z gpg: WARNING: This key is not certified with a trusted signature!
2024-11-30T10:08:04.525445235Z gpg:          There is no indication that the signature belongs to the owner.
2024-11-30T10:08:04.525479528Z Primary key fingerprint: 2880 6A87 8AE4 23A2 8372  792E D758 99B9 A724 937A
2024-11-30T10:08:04.526097753Z + mkdir -p /usr/src/tmp
2024-11-30T10:08:04.527736638Z + tar -xjf nextcloud.tar.bz2 -C /usr/src/tmp/
2024-11-30T10:08:59.851799949Z + gpgconf --kill all
2024-11-30T10:09:00.116868044Z + rm nextcloud.tar.bz2.asc nextcloud.tar.bz2
2024-11-30T10:09:00.226535149Z + mkdir -p /usr/src/tmp/nextcloud/data
2024-11-30T10:09:00.226686197Z + mkdir -p /usr/src/tmp/nextcloud/custom_apps
2024-11-30T10:09:00.231298919Z + chmod +x /usr/src/tmp/nextcloud/occ
2024-11-30T10:09:00.241266171Z + cp -r /usr/src/nextcloud/config/CAN_INSTALL /usr/src/nextcloud/config/aio.config.php /usr/src/nextcloud/config/apcu.config.php /usr/src/nextcloud/config/apps.config.php /usr/src/nextcloud/config/config.sample.php /usr/src/nextcloud/config/proxy.config.php /usr/src/nextcloud/config/redis.config.php /usr/src/nextcloud/config/reverse-proxy.config.php /usr/src/nextcloud/config/s3.config.php /usr/src/nextcloud/config/smtp.config.php /usr/src/nextcloud/config/swift.config.php /usr/src/tmp/nextcloud/config/
2024-11-30T10:09:00.276738922Z + mkdir -p /usr/src/tmp/nextcloud/apps/nextcloud-aio
2024-11-30T10:09:00.276819155Z + cp -r /usr/src/nextcloud/apps/nextcloud-aio/appinfo /usr/src/nextcloud/apps/nextcloud-aio/composer /usr/src/nextcloud/apps/nextcloud-aio/lib /usr/src/nextcloud/apps/nextcloud-aio/readme.md /usr/src/nextcloud/apps/nextcloud-aio/templates /usr/src/tmp/nextcloud/apps/nextcloud-aio/
2024-11-30T10:09:00.276865560Z + mv /usr/src/nextcloud /usr/src/temp-nextcloud
2024-11-30T10:09:00.446109534Z + mv /usr/src/tmp/nextcloud /usr/src/nextcloud
2024-11-30T10:09:00.460686966Z + rm -r /usr/src/tmp
2024-11-30T10:09:00.462260119Z + rm -r /usr/src/temp-nextcloud
2024-11-30T10:09:05.952406959Z ++ php -r 'require /usr/src/nextcloud/version.php; echo implode('\''.'\'', $OC_Version);'
2024-11-30T10:09:07.422229099Z + image_version='
2024-11-30T10:09:07.422583593Z Parse error: syntax error, unexpected token "/" in Command line code on line 1'
2024-11-30T10:09:20.691762396Z Connection to nextcloud-aio-database (172.18.0.3) 5432 port [tcp/postgresql] succeeded!
2024-11-30T10:09:23.057917491Z              now
2024-11-30T10:09:23.058334138Z ------------------------------
2024-11-30T10:09:23.058369182Z  2024-11-30 14:09:23.05725+04
2024-11-30T10:09:23.058402232Z (1 row)
2024-11-30T10:09:23.058428370Z
2024-11-30T10:09:23.121637853Z + '[' -f /dev-dri-group-was-added ']'
2024-11-30T10:09:23.121965369Z ++ find /dev -maxdepth 1 -mindepth 1 -name dri
2024-11-30T10:09:23.163495958Z + '[' -n '' ']'
2024-11-30T10:09:23.163621223Z + set +x
2024-11-30T10:09:23.260387193Z Connection to nextcloud-aio-redis (172.18.0.4) 6379 port [tcp/redis] succeeded!
2024-11-30T10:09:23.630625718Z The last Nextcloud update failed.
2024-11-30T10:09:23.630775680Z Please restore from backup and try again!
2024-11-30T10:09:23.630875184Z If you do not have a backup in place, you can simply delete the update.failed file in the datadir which will allow the container to start again.

And when upgrade continues after restart, notification was failed also

2024-11-30T11:16:10.992375753Z Posting notifications to users that are admins...
2024-11-30T11:16:15.906754771Z Posting 'Nextcloud update to 29.0.8.1 successful!' to: ncp
2024-11-30T11:16:16.385444856Z
2024-11-30T11:16:16.389520917Z
2024-11-30T11:16:16.389744086Z   The "--object-type" option does not exist.
2024-11-30T11:16:16.389900480Z
2024-11-30T11:16:16.389948526Z
2024-11-30T11:16:16.389973560Z notification:generate [-l|--long-message LONG-MESSAGE] [-d|--dummy] [--] <user-id> <short-message>
2024-11-30T11:16:16.390015312Z
2024-11-30T11:16:16.404878023Z Done!

**Comment by A11st0r** [comment]
I Think I updated from a 8.x version But I have the same Issue. 
"Parse error: syntax error, unexpected token "/" in Command line code on line 1'"

**Comment by RichEsq** [comment]
Having the same issue here

```
2024-12-06T01:11:18.877458350Z Connection to nextcloud-aio-database (172.22.0.4) 5432 port [tcp/postgresql] succeeded!
2024-12-06T01:11:20.984769110Z               now              
2024-12-06T01:11:20.984804793Z -------------------------------
2024-12-06T01:11:20.984810280Z  2024-12-06 12:11:20.973236+11
2024-12-06T01:11:20.984814877Z (1 row)
2024-12-06T01:11:20.984818982Z 
2024-12-06T01:11:21.002380330Z + '[' -f /dev-dri-group-was-added ']'
2024-12-06T01:11:21.002562091Z ++ find /dev -maxdepth 1 -mindepth 1 -name dri
2024-12-06T01:11:21.004047440Z + '[' -n '' ']'
2024-12-06T01:11:21.004075661Z + set +x
2024-12-06T01:11:21.039257333Z Enabling Imagick...
2024-12-06T01:11:22.864274042Z WARNING: opening from cache https://dl-cdn.alpinelinux.org/alpine/v3.20/main: No such file or directory
2024-12-06T01:11:22.864297409Z WARNING: opening from cache https://dl-cdn.alpinelinux.org/alpine/v3.20/community: No such file or directory
2024-12-06T01:11:22.916789608Z Installing PHP build dependencies...
2024-12-06T01:11:27.287751473Z Installing PHP extension tidy ...
2024-12-06T01:11:34.052873787Z configure: error: Cannot find libtidy
2024-12-06T01:11:34.080526293Z Could not install tidy from core. Trying to install from PECL...
2024-12-06T01:11:40.961175699Z error: 'tidy' does not exist
2024-12-06T01:11:40.961236569Z 
2024-12-06T01:11:40.961271386Z usage: /usr/local/bin/docker-php-ext-enable [options] module-name [module-name ...]
2024-12-06T01:11:40.961396383Z    ie: /usr/local/bin/docker-php-ext-enable gd mysqli
2024-12-06T01:11:40.961518899Z        /usr/local/bin/docker-php-ext-enable pdo pdo_mysql
2024-12-06T01:11:40.961633328Z        /usr/local/bin/docker-php-ext-enable --ini-name 0-apc.ini apcu apc
2024-12-06T01:11:40.961727247Z 
2024-12-06T01:11:40.961818060Z Possible values for module-name:
2024-12-06T01:11:41.022706831Z apcu.so bcmath.so bz2.so exif.so ftp.so gd.so gmp.so igbinary.so imagick.so imap.so intl.so ldap.so memcached.so opcache.so pcntl.so pdo_pgsql.so pgsql.so redis.so smbclient.so sodium.so sysvsem.so zip.so
2024-12-06T01:11:41.031447898Z 
2024-12-06T01:11:41.031590393Z Some of the above modules are already compiled into PHP; please check
2024-12-06T01:11:41.031709964Z the output of "php -i" to see which modules are already loaded.
2024-12-06T01:11:41.032060391Z Could also not install tidy from PECL. The PHP extensions was not installed!
2024-12-06T01:11:42.484306180Z WARNING: opening from cache https://dl-cdn.alpinelinux.org/alpine/v3.20/main: No such file or directory
2024-12-06T01:11:42.484439123Z WARNING: opening from cache https://dl-cdn.alpinelinux.org/alpine/v3.20/community: No such file or directory
2024-12-06T01:11:42.549673080Z Connection to nextcloud-aio-redis (172.22.0.5) 6379 port [tcp/redis] succeeded!
2024-12-06T01:11:42.854083201Z + NEXT_MAJOR=29
2024-12-06T01:11:42.854123210Z + curl -fsSL -o nextcloud.tar.bz2 https://download.nextcloud.com/server/releases/latest-29.tar.bz2
2024-12-06T01:12:02.236170201Z + curl -fsSL -o nextcloud.tar.bz2.asc https://download.nextcloud.com/server/releases/latest-29.tar.bz2.asc
2024-12-06T01:12:02.995231714Z ++ mktemp -d
2024-12-06T01:12:02.997390305Z + GNUPGHOME=/tmp/tmp.Odi2DfC6NM
2024-12-06T01:12:02.997419286Z + export GNUPGHOME
2024-12-06T01:12:02.997474394Z + gpg --batch --keyserver keyserver.ubuntu.com --recv-keys 28806A878AE423A28372792ED75899B9A724937A
2024-12-06T01:12:03.008167090Z gpg: keybox '/tmp/tmp.Odi2DfC6NM/pubring.kbx' created
2024-12-06T01:12:04.577425279Z gpg: /tmp/tmp.Odi2DfC6NM/trustdb.gpg: trustdb created
2024-12-06T01:12:04.577448263Z gpg: key D75899B9A724937A: public key "Nextcloud Security <security@nextcloud.com>" imported
2024-12-06T01:12:04.584359087Z gpg: Total number processed: 1
2024-12-06T01:12:04.584380648Z gpg:               imported: 1
2024-12-06T01:12:04.585095868Z + gpg --batch --verify nextcloud.tar.bz2.asc nextcloud.tar.bz2
2024-12-06T01:12:05.005289574Z gpg: Signature made Thu Dec  5 23:29:17 2024 AEDT
2024-12-06T01:12:05.005370975Z gpg:                using RSA key 28806A878AE423A28372792ED75899B9A724937A
2024-12-06T01:12:05.006358059Z gpg: Good signature from "Nextcloud Security <security@nextcloud.com>" [unknown]
2024-12-06T01:12:05.006371078Z gpg: WARNING: This key is not certified with a trusted signature!
2024-12-06T01:12:05.006376835Z gpg:          There is no indication that the signature belongs to the owner.
2024-12-06T01:12:05.006382026Z Primary key fingerprint: 2880 6A87 8AE4 23A2 8372  792E D758 99B9 A724 937A
2024-12-06T01:12:05.006974244Z + mkdir -p /usr/src/tmp
2024-12-06T01:12:05.009019767Z + tar -xjf nextcloud.tar.bz2 -C /usr/src/tmp/
2024-12-06T01:12:44.044229505Z + gpgconf --kill all
2024-12-06T01:12:44.058316001Z + rm nextcloud.tar.bz2.asc nextcloud.tar.bz2
2024-12-06T01:12:44.094381919Z + mkdir -p /usr/src/tmp/nextcloud/data
2024-12-06T01:12:44.097273533Z + mkdir -p /usr/src/tmp/nextcloud/custom_apps
2024-12-06T01:12:44.099250371Z + chmod +x /usr/src/tmp/nextcloud/occ
2024-12-06T01:12:44.103110003Z + cp -r /usr/src/nextcloud/config/CAN_INSTALL /usr/src/nextcloud/config/aio.config.php /usr/src/nextcloud/config/apcu.config.php /usr/src/nextcloud/config/apps.config.php /usr/src/nextcloud/config/config.sample.php /usr/src/nextcloud/config/proxy.config.php /usr/src/nextcloud/config/redis.config.php /usr/src/nextcloud/config/reverse-proxy.config.php /usr/src/nextcloud/config/s3.config.php /usr/src/nextcloud/config/smtp.config.php /usr/src/nextcloud/config/swift.config.php /usr/src/tmp/nextcloud/config/
2024-12-06T01:12:44.110015829Z + mkdir -p /usr/src/tmp/nextcloud/apps/nextcloud-aio
2024-12-06T01:12:44.114298382Z + cp -r /usr/src/nextcloud/apps/nextcloud-aio/appinfo /usr/src/nextcloud/apps/nextcloud-aio/composer /usr/src/nextcloud/apps/nextcloud-aio/lib /usr/src/nextcloud/apps/nextcloud-aio/readme.md /usr/src/nextcloud/apps/nextcloud-aio/templates /usr/src/tmp/nextcloud/apps/nextcloud-aio/
2024-12-06T01:12:44.124971628Z + mv /usr/src/nextcloud /usr/src/temp-nextcloud
2024-12-06T01:12:56.932107109Z + mv /usr/src/tmp/nextcloud /usr/src/nextcloud
2024-12-06T01:12:56.938880345Z + rm -r /usr/src/tmp
2024-12-06T01:12:56.941012490Z + rm -r /usr/src/temp-nextcloud
2024-12-06T01:12:57.885353706Z ++ php -r 'require /usr/src/nextcloud/version.php; echo implode('\''.'\'', $OC_Version);'
2024-12-06T01:12:58.203203200Z + image_version='
2024-12-06T01:12:58.203228067Z Parse error: syntax error, unexpected token "/" in Command line code on line 1'
2024-12-06T01:12:59.053839906Z Connection to nextcloud-aio-database (172.22.0.4) 5432 port [tcp/postgresql] succeeded!
2024-12-06T01:13:01.094912621Z               now    
```

**Comment by szaimen** [comment]
> php -r 'require /usr/src/nextcloud/version.php; echo implode('''.''', $OC_Version);'
> 2024-11-30T10:09:07.422229099Z + image_version='
> 2024-11-30T10:09:07.422583593Z Parse error: syntax error, unexpected token "/" in Command line code on line 1'

This seems to be the problem

**Comment by szaimen** [comment]
Hi, thanks for the report! A fix is in https://github.com/nextcloud/all-in-one/pull/5718

**Comment by szaimen** [comment]
This is now fixed with v10.1.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel
